luaui/Widgets/map_startbox.lua

```diff
@@ -266,6 +225,12 @@ local LuaShader = VFS.Include(luaShaderDir.."LuaShader.lua")
 VFS.Include(luaShaderDir.."instancevbotable.lua")
 
 local minY, maxY = Spring.GetGroundExtremes()
+local waterlevel = (Spring.GetModOption and Spring.GetModOptions().map_waterlevel) or 0
+if waterlevel > 0 then
```

Comment:
waterlevel can be negative, anyways I think you can use the 3rd and 4th parameters from [GetGroundExtremes](https://github.com/beyond-all-reason/spring/blob/0c3689c3678219f529eb965f4aae91627d1b98ee/rts/Lua/LuaSyncedRead.cpp#L7521), those give you currMinY and currMaxY as opposed to initial values.

Reply:
actually, forget that, at least if you get those here, it's too early for currMinY and currMaxY to be set since terraforming happens later.

still should account for negative waterlevel tho.

Reply:
Btw, There is some (broken) code to re-init when the waterlevel changes [here](https://github.com/beyond-all-reason/Beyond-All-Reason/blob/14c64a8a68b58aad284a3e61e777c3bc33062636/luaui/Widgets/map_startbox.lua#L614) and [here](https://github.com/beyond-all-reason/Beyond-All-Reason/blob/14c64a8a68b58aad284a3e61e777c3bc33062636/luaui/Widgets/map_startbox.lua#L629).

I think you can safely remove all that since looks like it was utterly broken anyways.

Also this is a widget that gets autoremoved when the game starts, so probably not worth it all that hassle of trying to refresh if the waterlevel changes like that.